cibyl/outputs/cli/printer.py

```diff
@@ -102,7 +104,54 @@ def palette(self) -> ColorPalette:
         return self._palette
 
 
-class SerializedPrinter(Printer, ABC):
+class SerializationProvider:
+    @dataclass
+    class Functions:
+        load: Callable[[str], dict]
+        """Transforms machine-readable text into a python structure."""
+        dump: Callable[[dict], str]
+        """Transforms a python structure into machine-readable text."""
+
+    def __init__(self, functions: Functions):
+        self._functions = functions
+
+    @property
+    def load(self) -> Callable[[str], dict]:
+        return self._functions.load
+
+    @property
+    def dump(self) -> Callable[[dict], str]:
+        return self._functions.dump
+
+
+PROV = TypeVar('PROV', bound=SerializationProvider)
+
+
+class JSON(SerializationProvider, ABC):
+    @property
+    @abstractmethod
+    def indentation(self) -> int:
+        raise NotImplementedError
+
+
+class STDJSON(JSON):
```

Comment:
What are the possbile use cases for having a `JSON` and a child `STDJSON`class? In case we want to use a different json library?

Reply:
That is the idea. Now that I had the chance I made it as future proof as I could.

Reply:
Makes sense, thanks!